delocate/libsana.py

```diff
@@ -104,6 +113,56 @@ def resolve_rpath(lib_path, rpaths):
     return lib_path
 
 
+def search_environment_for_lib(lib_path):
+    """ Search common environment variables for `lib_path`
+
+    Apple's documentation about how to do this does not quite match what we see
+    in the wild, so we'll use a single approach here:
+
+        1. Search for the basename of the library on DYLD_LIBRARY_PATH
+        2. Search for ``realpath(lib_path)``
+        3. Search for the basename of the library on DYLD_FALLBACK_LIBRARY_PATH
+
+    Testing has suggested that LD_LIBRARY_PATH is not actually used in this
+    process as all, so we'll ignore it.
+
+    Parameters
+    ----------
+    lib_path : str
+        Name of the library to search for
+
+    Returns
+    -------
+    lib_path : str
+        Full path of ``basename(lib_path)``'s location, if it can be found, or
+        ``realpath(lib_path)`` if it cannot.
+    """
+    lib_basename = basename(lib_path)
+    potential_library_locations = []
+
+    def _paths_from_var(varname, lib_basename):
+        var = os.environ.get(varname)
+        if var is None:
+            return []
+        return [pjoin(path, lib_basename) for path in var.split(':')]
+
+    # 1. Search on DYLD_LIBRARY_PATH
+    potential_library_locations += _paths_from_var('DYLD_LIBRARY_PATH',
+                                                   lib_basename)
+
+    # 2. Search for realpath(lib_path)
```

Comment:
Add comment here to say why `realpath` is necessary?

Reply:
See my comment above and the `testing_osx_rpath_env_variables.sh` script.